frontend/src/Editor/Components/Button.jsx

```diff
@@ -51,7 +49,7 @@ export const Button = function Button({ id, width, height, component, onComponen
         onComponentClick(id, component);
       }}
     >
-      {text}
+      {getDynamicText(text, currentState)}
```

Comment:
You can directly invoke `resolveReferences` here, instead of calling a separate function to do so.
```suggestion
      {resolveReferences(text, currentState)}
```